src/core/EdFi.DataManagementService.Core/Middleware/ValidateRepeatedPropertiesMiddleware.cs

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Globalization;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using EdFi.DataManagementService.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+using EdFi.DataManagementService.Core.Response;
+using Microsoft.Extensions.Logging;
+using static EdFi.DataManagementService.Core.Response.FailureResponse;
+
+namespace EdFi.DataManagementService.Core.Middleware;
+
+internal class ValidateRepeatedPropertiesMiddleware(ILogger logger) : IPipelineStep
+{
+    private static readonly JsonSerializerOptions _serializerOptions =
+        new() { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping };
+
+    public async Task Execute(PipelineContext context, Func<Task> next)
+    {
+        logger.LogDebug(
+            "Entering ValidateRepeatedPropertiesMiddleware - {TraceId}",
+            context.FrontendRequest.TraceId
+        );
+        if (context.FrontendRequest.Body != null)
+        {
+            try
+            {
+                var jsonDocument = JsonDocument.Parse(context.FrontendRequest.Body);
+                var validationErrors = new Dictionary<string, List<string>>();
+
+                CheckForDuplicateProperties(jsonDocument.RootElement, "", validationErrors);
+
+                CheckForDuplicateValues(jsonDocument.RootElement, "", validationErrors);
```

Comment:
Are we iterating over the entire document twice?

Reply:
Is there any way to use JSON Schema to block duplicate properties, since it's already scanning the whole document?

Reply:
Basically, try to avoid that a data collection contains 2 or more records with the same values; this is currently being validated by ODS.

Reply:
> Are we iterating over the entire document twice?

Consolidated all in one iteration.